### PR TITLE
Move SBS handler to AddMemberTx, add more SBS tests

### PR DESCRIPTION
**Pull request by zapu**
1. Increase SBS handler test coverage. New tests exercise SBS canceling mooted invite code path (see 2), and also sweeping old versions of members when adding users through SBS.
2. Discover dead code path where SBS handler cancels social invite for which the user is already member of the team with same role or higher. This does not trigger anymore because server will automatically set invite status to MOOTED and not trigger SBS handler in clients.
3. Use AddMemberTx in SBS handler. This required a new function in `transactions.go`: `AddMemberBySBS` which is specific to team invite resolution.
4. Changes to `AddMemberTx.sweepKeybaseInvites` were needed so it could work with `CompleteInviteByID` and not try to cancel already completed invite.

### CODE REVIEW

**Comment by mlsteele** [comment]
2. We want to cancel the invite, right? Hanging invites are an unnecessary weakness. If we've talked about this before and I forgot, then that's fine.

**Comment by zapu** [comment]
Would you mind if we had another ticket to deal with 2? It's a bit confusing and I don't think I have the full picture.

**Comment by mlsteele** [comment]
Sure that sounds good

File: go/teams/transactions.go [hunk 1]
```diff
@@ -420,6 +437,57 @@ func (tx *AddMemberTx) ReAddMemberToImplicitTeam(ctx context.Context, uv keybase
 	return nil
 }
 
+// AddMemberBySBS is very similar in what it does to addMemberByUPAKV2
+// (or AddMemberBy* family of functions), but it has easier job
+// because it only adds cryptomembers and fails on PUKless users. It
+// also completed inviteID for given invitee.
```
Comment (mlsteele) [inline go/teams/transactions.go]: `It also completed inviteID for given invitee.`
What did you mean?

Comment (zapu) [reply go/teams/transactions.go]: It signs in `completedInvites[invitee.inviteID] = uv` to the team chain. Let me rephrase that to something better.
